src/NiceHashMinerLegacy/Miners/MiningSession.cs

```diff
@@ -275,6 +275,31 @@ public void UpdateUsedDevices(IEnumerable<ComputeDevice> devices)
             _switchingManager.Start();
         }
 
+        private void RestartRunningGroupMiners()
+        {
+            foreach (var key in _runningGroupMiners.Keys)
+            {
+                _runningGroupMiners[key].Stop();
+                _runningGroupMiners[key].Start(_miningLocation, _btcAdress, _worker);
+            }
+        }
+
+        public void UpdateBTC(string btc)
+        {
+            _switchingManager.Stop(); // TODO NOT SURE?
```

Comment:
@DillonN please take note of the `// TODO NOT SURE?` comments in this file. I am not sure if the timer needs to be stopped and resumed.

Reply:
Yes, `_switchingManager` has the potential to call `SwichMostProfitableGroupUpMethod()` in a callback when running, which may change `_runningGroupMiners` on a separate thread causing an `InvalidOperationException` in the `foreach`